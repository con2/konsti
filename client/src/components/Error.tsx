import React, { FC, ReactElement } from 'react';

interface Props {
  meta: { touched: boolean; error: string };
}

export const Error: FC<Props> = (props: Props): ReactElement => {
  const {
    meta: { touched, error },
  } = props;

  return touched && error ? <span>{error}</span> : <span />;
};
