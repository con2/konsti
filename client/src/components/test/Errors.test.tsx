import React from 'react';
import { shallow } from 'enzyme';
import { Error } from '../Error';

describe('Errors', () => {
  const meta = { touched: true, error: 'error' };

  it('should render correctly', () => {
    const component = shallow(<Error meta={meta} />);
    expect(component).toMatchSnapshot();
  });
});
