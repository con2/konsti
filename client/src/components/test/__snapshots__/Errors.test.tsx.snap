// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors should render correctly 1`] = `
<span>
  error
</span>
`;
